framework/src/loops/ComputeElemAuxVarsThread.C

```diff
@@ -112,8 +115,27 @@ ComputeElemAuxVarsThread<AuxKernelType>::onElement(const Elem * elem)
       _fe_problem.reinitMaterials(elem->subdomain_id(), _tid);
 
     for (const auto & aux : kernels)
+    {
       aux->compute();
 
+      // update the aux solution vector if writable coupled variables are used
+      if (aux->hasWritableCoupledVariables())
+      {
+        Threads::spin_mutex::scoped_lock lock(writable_variable_mutex);
+        for (auto * var : aux->getWritableCoupledVariables())
+        {
+          // insert into the global solution vector
+          var->insert(_aux_sys.solution());
+          var->prepareAux();
+        }
+
+        // make solution values available for dependent AuxKernels
+        aux->variable().insert(_aux_sys.solution());
```

Comment:
With this change and with the following `reinitElem` call, I'm hoping there is no need to `_var.setNodalValue` in `AuxKernel::compute` anymore. We should have either one or the other.

Reply:
Except this is currently conditional on `hasWritableCoupledVariables`, but I suppose I could pull it out and remove teh call in AuxKernel. Thanks for the review by the way, better late than never :-D

Reply:
I'm only reviewing because I'm modifying these loops right now and there is all this new code which I'm trying to understand the motivation for ... so I figured the best place to comment and ask my questions would be where the code was originally added